Libs/vtkSegmentationCore/vtkSegmentation.cxx

```diff
@@ -271,18 +280,59 @@ bool vtkSegmentation::SetSegmentModifiedEnabled(bool enabled)
 }
 
 //---------------------------------------------------------------------------
-std::string vtkSegmentation::GenerateUniqueSegmentID(std::string id)
+std::string vtkSegmentation::GenerateUniqueSegmentName(std::string base)
 {
-  if (!id.empty() &&  this->Segments.find(id) == this->Segments.end())
+  if (base.empty())
+  {
+    base = "Segment";
```

Comment:
The segment name is displayable on the GUI, so it should be translatable. Since order of words and numbers are language specific, instead of using a `base` string, it would be better to pass a `pattern = vtkMRMLTr("vtkSomeClassName", "Segment_%1")`.

Inside this method we should use `vtkMRMLI18N::Format(pattern, nameIndexStr)` to replace the `%1` placeholder. However, this library does not depend on MRML, so maybe we would need to copy `vtkMRMLI18N::Format` method into this class (maybe not the complete method, just a version that handles one placeholder). Maybe we could move `vtkMRMLI18N::Format` to `vtkAddon` with some more generic name like `vtkAddonStringUtilities::FormatString(...)`.

Reply:
I don't think we can use vtkMRMLTr either unless we want to move all of the translation functions into vtkAddon.

Reply:
If that makes things easier, we can move `vtkMRMLI18N` to `vtkAddon`. We would then need to rename to `vtkI18N` or maybe `vtkInternationalization`. It is a singleton class, so maybe we should move the singleton macros to `vtkAddon`, too?